llvm/lib/Target/NVPTX/NVPTXUtilities.cpp

```diff
@@ -311,11 +311,16 @@ std::optional<unsigned> getMaxNReg(const Function &F) {
 }
 
 bool isKernelFunction(const Function &F) {
+  if (F.getCallingConv() == CallingConv::PTX_Kernel)
+    return true;
+
+  if (F.hasFnAttribute("nvvm.kernel"))
+    return true;
+
   if (const auto X = findOneNVVMAnnotation(&F, "kernel"))
     return (*X == 1);
 
-  // There is no NVVM metadata, check the calling convention
```

Comment:
Is there a reason we're ignoring calling convention as an indication of the function being a kernel?

I think standalone compilation may be relying on that. @jhuber6 -- is this needed for your use case?

Reply:
Unsure, this code seems to be very old. If it breaks anything ideally the `libc` build bot would complain.

Reply:
We still have the calling convention check, I just moved it higher up in the function since it is cheaper than traversing the metadata.